frontends/PyCDE/src/pycde/value.py

```diff
@@ -43,13 +44,19 @@ def __new__(cls, value, type=None):
 
   _reg_name = re.compile(r"^(.*)__reg(\d+)$")
 
-  def reg(self, clk, rst=None, name=None, cycles=1, sv_attributes=None):
+  def reg(self, clk=None, rst=None, name=None, cycles=1, sv_attributes=None):
     """Register this value, returning the delayed value.
     `clk`, `rst`: the clock and reset signals.
     `name`: name this register explicitly.
     `cycles`: number of registers to add."""
+
+    if clk is None:
+      clk = ClockValue._get_current_clock_block()
+      if clk is None:
+        raise ValueError("If 'clk' not specified, must be in clock block")
```

Comment:
Would you mind adding a test to catch this erroring behaviour?

Reply:
https://github.com/llvm/circt/commit/2b8bd3b26d6dafe9752b4dac1a536bd2478b2a78